scripts/scil_filter_tractogram_anatomically.py

```diff
@@ -94,21 +168,36 @@ def main():
     if args.verbose:
         logging.basicConfig(level=logging.DEBUG)
 
-    if args.minL == 0 and np.isinf(args.maxL):
-        logging.debug("You have not specified minL nor maxL. Output will "
-                      "not be filtered according to length!")
-
     if args.angle <= 0:
         parser.error('Angle "{}" '.format(args.angle) +
                      'must be greater than 0')
 
+    sft = load_tractogram_with_reference(parser, args, args.in_tractogram)
+
+    img_wmparc = nib.load(args.in_wmparc)
+    if not is_header_compatible(img_wmparc, sft):
+        parser.error('Headers from the tractogram and the wmparc are '
+                     'not compatible.')
+
+    if args.csf_bin:
+        img_csf = nib.load(args.csf_bin)
+        if not is_header_compatible(img_csf, sft):
+            parser.error('Headers from the tractogram and the CSF mask are '
+                        'not compatible.')
+
+    if args.minL == 0 and np.isinf(args.maxL):
+        logging.debug("You have not specified minL nor maxL. Output will "
+                      "not be filtered according to length!")
+
     if np.isinf(args.angle):
         logging.debug("You have not specified the angle. Loops will "
                       "not be filtered!")
 
```

Comment:
A check is needed to ensure that `ctx_dilation_radius > 0`. If 0, you may print a message saying that no dilation will be performed.

Reply:
Done in dbd0b2a.